### PR TITLE
perf(ascend): refactor aclnn adaptor & add copy & fix storage shape

### PR DESCRIPTION
**Pull request by lljbash**
Thanks for your contribution and we appreciate it a lot. The following instructions would make your pull request more healthy and more easily get feedback. If you do not understand some items, don't worry, just make the pull request and seek help from maintainers.

## Description

1. simplify and speedup aclnn adaptor, rename API
2. speedup AcsendTensor ctor a bit
3. refactor code structure of AcsendTensor a bit for aclnn adaptor
4. add aclnn version of copy
5. ~~disable copy test for float64~~
6. migrate existing aclnn ops to the new adaptor

## Checklist
**Before PR**:

- [x] I have read and followed the workflow indicated in the [Contributors.md](https://github.com/DeepLink-org/DIOPI/blob/main/Contributors.md) to create this PR.
- [x] Pre-commit or linting tools indicated in [Contributors.md](https://github.com/DeepLink-org/DIOPI/blob/main/Contributors.md) are used to fix the potential lint issues.
- [x] Bug fixes are covered by unit tests, the case that causes the bug should be added in the unit tests.
- [x] New functionalities are covered by complete unit tests. If not, please add more unit test to ensure the correctness.
- [x] The documentation has been modified accordingly, including docstring or example tutorials.

**After PR**:

- [x] CLA has been signed and all committers have signed the CLA in this PR.

### CODE REVIEW

File: impl/ascend/aclnn/adaptor.hpp [hunk 1]
```diff
@@ -57,98 +44,127 @@ inline void* getOpApiLibHandler(const char* libName) {
...
-inline aclTensor* convertType(AclTensor& value) { return static_cast<aclTensor*>(value); }
+template <class... Args>
+decltype(auto) callOpApiFuncWithConvertedParams(void* opApiAddr, Args&&... args) {
+    return callOpApiFunc(opApiAddr, convertType(std::forward<Args>(args))...);
+}
 
-template <typename T>
-T convertType(T value) {
-    return value;
+inline void logDebugIfEnabled(const char* api) {
```
Comment (yangbofun) [inline impl/ascend/aclnn/adaptor.hpp:123]: 有一个统一管理环境变量的地方(env_var.h和env_var.cpp)，可以放到这里。

File: impl/ascend/aclnn/adaptor.hpp [hunk 2]
```diff
@@ -57,98 +44,127 @@ inline void* getOpApiLibHandler(const char* libName) {
...
+    if (aclDebugFlag) {
+        std::cout << "ACLNN_ADAPTOR for " << api << '\n';
+    }
 }
 
-template <typename... Ts>
-constexpr auto convertTypes(Ts&... args) {
-    return std::make_tuple(convertType(args)...);
+template <class... Args>
+void callAclnnImpl(const char* api, const char* workspaceApi, diopiContextHandle_t ctx, Args&&... args) {
```
Comment (yangbofun) [inline impl/ascend/aclnn/adaptor.hpp:131]: 看下面的宏DIOPI_ASCEND_CALL_ACLNN，workspaceApi写死的是“ GetWorkspaceSize”，这里是否可以不作为传参，在函数内写死这个值。

File: impl/ascend/ascend_tensor.cpp [hunk 3]
```diff
@@ -167,17 +168,17 @@ int64_t AscendTensor::getAclMemBufferSize() const {
     }
 }
 
-aclFormat AscendTensor::getAclDataFormat() const {
+aclFormat inferAclDataFormat(int64_t dim, const int64_t* shape, const int64_t* stride, diopiConstTensorHandle_t tensor) {
```
Comment (yangbofun) [inline impl/ascend/ascend_tensor.cpp:171]: memory format应该只需要shape和stride，这里的tensor参数应该可以删掉。

File: impl/ascend/aclnn/adaptor.hpp (hunk 2 above)
Comment (lljbash) [reply impl/ascend/aclnn/adaptor.hpp:131]: 编译期 strcat 用宏做比较容易，如果不想要一个巨大的宏，就得这么传，我可以加段注释

File: impl/ascend/ascend_tensor.cpp (hunk 3 above)
Comment (lljbash) [reply impl/ascend/ascend_tensor.cpp:171]: 这个 tensor 用来 debug 的，参见声明处的注释
是否要把这个 debug 删除掉？

Comment (lljbash) [reply impl/ascend/ascend_tensor.cpp:171]: 或许可以重写这个报错逻辑，只输出 shape 和 stride

File: impl/ascend/aclnn/adaptor.hpp (hunk 2 above)
Comment (yangbofun) [reply impl/ascend/aclnn/adaptor.hpp:131]: ok

**Review by yangbofun — APPROVED** [review]

**Pull request merged**